src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -92,6 +92,8 @@ const std::shared_ptr<AbstractLQPNode>& SQLPipelineStatement::get_unoptimized_lo
   DebugAssert(lqp_roots.size() == 1, "LQP translation returned no or more than one LQP root for a single statement.");
   _unoptimized_logical_plan = lqp_roots.front();
 
+  _cacheable = sql_translator.cacheable();
```

Comment:
Convenient, but I am not sure if this is really a property/responsibility of the SQLTranslator. Another option would be to have this property stored in the StoredTableNode, but then we would have to traverse each result and check for StoredTableNodes, which feels like pointless work. :/

Reply:
indeed. another option would be to put the flag in the AbstractLQPNode and set it for the root node if the whole LQP contains a metatable access. this way, neither the sql_translator nor the sql_pipelined_statement would need a member variable. what do you think about this?

Reply:
By design, LQP nodes shouldn't statically hold any information about their leafs. If they did, e.g., hold the order of expressions, LQP manipulations below the node would become a pain.

In this case, doing so would probably be harmless, but I don't want to open Pandora's box.

Reply:
our discussed solution of passing the parser result to the SQLTranslator constructor isn't that straight forward since other instances of the SQLTranslator are sometimes used within the initial sql translater in order to translate other kinds of SQL statements that don't need any sqlParserResults. see:
https://github.com/j-tr/hyrise/blob/06a5f900999a497c30a96dd5e9bb8f79e60f2627/src/lib/sql/sql_translator.cpp#L288
https://github.com/j-tr/hyrise/blob/06a5f900999a497c30a96dd5e9bb8f79e60f2627/src/lib/sql/sql_translator.cpp#L308
I think the with_translator could be omitted and the method could be called directly instead but the one in translate_hsql_expr needs to have an empty context. Do you have an idea how to solve this?

Reply:
Hmpf . :/

in that case, returning a pair with TranslationInfo might be easier. Can we make the ID thing used by the server part of that, too?

Reply:
i now implemented a solution that avoids modifying the return types of all translate_* functions by still using a private member variable _cacheable but without having a public getter for it. This way, from an external view, the SQL translator is stateless